### PR TITLE
Alphanumeric Callsign For Flights

### PR DESCRIPTION
**Pull request by FatihKoz**
PR adds ability to define alphanumeric callsigns for flights, display them and use them during SimBrief OFP generation (if pilot ident as callsign option is not enabled)

Translations for en/es/it/pt-br added.

### CODE REVIEW

File: app/Database/migrations/2021_04_05_055245_flights_add_alphanumeric_callsign.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * Exp DLH78BF, THY8EA, OGE1978
+ * According to FAA and EASA, callsigns must be maximum 7 chars in which first 3 chars is
+ * airline ICAO code remaining rest can be used freely according to airline's choices
+ */
+class FlightsAddAlphanumericCallsign extends Migration
+{
+    public function up()
+    {
+        Schema::table('flights', function (Blueprint $table) {
+            $table->string('anumeric_callsign', 4)
```
Comment (nabeelio) [inline app/Database/migrations/2021_04_05_055245_flights_add_alphanumeric_callsign.php:18]: I think just call this `callsign`

File: app/Models/Flight.php [hunk 2]
```diff
@@ -60,6 +61,7 @@ class Flight extends Model
         'id',
         'airline_id',
         'flight_number',
+        'anumeric_callsign',
```
Comment (nabeelio) [inline app/Models/Flight.php:64]: `callsign`

**Comment by nabeelio** [comment]
In the model, the `getIdentAttribute` method should return the callsign if the `callsign` field is not empty.

File: resources/views/layouts/default/flights/simbrief_form.blade.php [hunk 3]
```diff
@@ -143,6 +143,8 @@
               <input type="hidden" name="fltnum" value="{{ $flight->flight_number }}">
               @if(setting('simbrief.callsign', true))
                 <input type="hidden" name="callsign" value="{{ $user->ident }}">
+              @else
+                <input type="hidden" name="callsign" value="{{ $flight->airline->icao }}{{ $flight->anumeric_callsign ?? $flight->flight_number }}">
```
Comment (nabeelio) [inline resources/views/layouts/default/flights/simbrief_form.blade.php:147]: Probably better to just use `$flight->ident` here.

File: app/Models/Flight.php [hunk 4]
```diff
@@ -104,6 +106,7 @@ class Flight extends Model
     public static $rules = [
         'airline_id'           => 'required|exists:airlines,id',
         'flight_number'        => 'required',
+        'anumeric_callsign'    => 'nullable',
```
Comment (nabeelio) [inline app/Models/Flight.php:109]: This should be `string|max:4|nullable`.

**Comment by FatihKoz** [comment]
> In the model, the `getIdentAttribute` method should return the callsign if the `callsign` field is not empty. Unless `callsign` should be treated separately.

Doesn't the flight ident is something like THY1978/Cnnnnn/Lyy when code and leg fields are not null ?

If this is the case, then using Callsign in that ident would not be good to track/display flights, specially if we think that callsign is only used for ATC communications, i do not think that we can use $flight->ident at simbrief too.

But if the $flight->ident is only THY1978 always (regardless of code/leg) , then yes we can use it directly at simbrief but then the blades used to display flight info should not use it 'cause it is really hard to memorize alphanumeric callsigns and they make no sense when looking to flights table/details etc.

**Comment by nabeelio** [comment]
Yes, `ident` does include the code/leg. If simbrief is the only place it'd be used then that's fine to leave it as-is.

**Comment by FatihKoz** [comment]
Probably it will be used at SimBrief only to generate flight plans/atc messages for IVAO/VATSIM etc. Not something we would print out on boarding passes. ( will handle other changes soon )

**Comment by FatihKoz** [comment]
All done, i hope ...

I did not changed or messed up with csv import though, it would be nice to have callsign field available for csv import too.

**Comment by nabeelio** [comment]
CSV import should be unchanged, there are tests for that and seems they're passing.

**Pull request merged**